x-pack/plugins/fleet/public/applications/fleet/sections/agent_policy/edit_package_policy_page/index.tsx

```diff
@@ -150,7 +159,21 @@ export const EditPackagePolicyForm = memo<{
           packagePolicyId,
         ]);
 
-        if (upgradePackagePolicyDryRunData) {
+        const hasUpgradeAvailable =
+          upgradePackagePolicyDryRunData &&
+          upgradePackagePolicyDryRunData[0].diff &&
+          semverGt(
```

Comment:
nit: you could move this logic to a small function to be less repetitive e.g. `hasVersionDiff(upgradePackagePolicyDryRunData[0].diff)`

Reply:
Good call - I moved this to a `utils/has_upgrade_available.ts` module in this directory.